main.js

```diff
@@ -172,21 +172,21 @@ async function install_plain_lua_windows(luaExtractPath, luaInstallPath, luaVers
     }
   })
 
-  objs["lua"] = [ ...objs["lua"], ...objs["lib"] ]
-  objs["luac"] = [ ...objs["luac"], ...objs["lib"] ]
-
   let luaXYZ = luaVersion.split(".")
   let libFile = "lua" + luaXYZ[0] + luaXYZ[1] + ".lib"
   let dllFile = "lua" + luaXYZ[0] + luaXYZ[1] + ".dll"
 
+  objs["lua"] = [ ...objs["lua"], libFile ]
+  objs["luac"] = [ ...objs["luac"], ...objs["lib"] ]
+
   await msvc_link(luaExtractPath, "link /nologo /DLL", dllFile, objs["lib"]);
   await msvc_link(luaExtractPath, "link /nologo", "luac.exe", objs["luac"]);
   await msvc_link(luaExtractPath, "link /nologo", "lua.exe", objs["lua"]);
 
   const luaHpp = (await exists(pathJoin(src, "lua.hpp"))) ? "lua.hpp" : "../etc/lua.hpp"
   const headers = [ "lua.h", "luaconf.h", "lualib.h", "lauxlib.h", luaHpp ]
 
-  await install_files(pathJoin(luaInstallPath, "bin"), luaExtractPath, [ "lua.exe", "luac.exe" ])
+  await install_files(pathJoin(luaInstallPath, "bin"), luaExtractPath, [ "lua.exe", "luac.exe", dllFile ])
   await install_files(pathJoin(luaInstallPath, "lib"), luaExtractPath, [ dllFile, libFile ])
```

Comment:
I think the dll should be installed in only one place, not in both `lib` and `bin`:
```suggestion
  await install_files(pathJoin(luaInstallPath, "bin"), luaExtractPath, [ "lua.exe", "luac.exe", dllFile ])
  await install_files(pathJoin(luaInstallPath, "lib"), luaExtractPath, [ libFile ])
```

Reply:
Copying the file in both places won't hurt (it's a small file in a temporary CI environment), and I can't tell for certain that _not_ copying the file to `lib` won't break someone else's flows which are relying on that file being there.

What I do know is that this PR as is fixes the C module loading behavior. Can we merge this as-is? Let's not get this held back on whether it produces a redundant file.

Reply:
Sorry for the late reply, must have overseen this. @hishamhm Agreed, I believe to remember seeing some linking problems with some rocks when the file was not present in lib, I think it was related to stub-libs on Windows / MSVC.